sapp/tests/warning_messages_test.py

```diff
@@ -0,0 +1,84 @@
+import json
+import pathlib
+from unittest import TestCase
+
+from sqlalchemy import Table
+
+from ..db import DB, DBType
+from ..models import WarningMessage, create as create_model
+from ..warning_messages import update_warning_messages, upsert_entry
+from .fake_object_generator import FakeObjectGenerator
+
+test_metadata = {
+    "codes": {
+        "1001": "Updated warning message from test",
+        "1002": "Updated warning message from test 2",
+    },
+    "codes2": {
+        "2001": "Updated warning message from another test",
+        "2002": "Updated warning message from another test 2",
+    },
+}
+
+
+class WarningMessagesTest(TestCase):
+    def setUp(self) -> None:
+        self.db = DB(DBType.MEMORY)
+        create_model(self.db)
+        self.fakes = FakeObjectGenerator()
+
+        with open("sample_metadata.json", "w") as outfile:
+            json.dump(test_metadata, outfile)
+
+    def tearDown(self) -> None:
+        pathlib.Path("sample_metadata.json").unlink()
+
+    def test_update_warning_messages(self) -> None:
+        with self.db.make_session() as session:
+            update_warning_messages(self.db, pathlib.Path("sample_metadata.json"))
+            self.assertEqual(
+                session.query(WarningMessage)
+                .filter(WarningMessage.code == "1001")
+                .first()
+                .__dict__["message"],
```

Comment:
Why is this using the `__dict__` method rather then just access the attribute?
```suggestion
                .message,
```

Reply:
Actually, it was giving a type error when I was using that (in the previous commit).
```
sapp/tests/warning_messages_test.py:40:16 Undefined attribute [16]: Optional type has no attribute `message`.
```

Reply:
Pyre is probably unhappy because [`sqlalchemy.orm.Query.first()` can return `None`](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.first) if no row is found with the 1001 warning code. 

I think we might have a few options here, but not sure if others have an opinion on the preferred way: 
1. We can write a test case before this one that [checks for the existence of the row](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.exists) with warning code 1001, then change `first()` to [`one_or_none()`](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.one_or_none). That way when this test case runs and 1001 doesn't exist in the table, the previous test case would fail before the exception from `one_or_none()` is thrown.
2. Switch `first()` to `one_or_none()` in this test case and treat the thrown exception as a failed test

Reply:
Did the first one @0xedward